File: src/main/java/no/nav/pensjon/innsyn/domain/desc/OpptjeningDescriptor.java
package no.nav.pensjon.innsyn.domain.desc;

import no.nav.pensjon.innsyn.entity.EntityDescriptor;

public class OpptjeningDescriptor implements EntityDescriptor {

    private static final String ENTITY_NAME = "Opptjening";

    private static final String[] PROPERTY_NAMES = {
            "Type",
            "Status",
            "Opptjeningsår",
            "Pensjonsgivende inntekt",
            "Poeng",
            "Uføregrad"
    };

    public String getEntityName() {
        return ENTITY_NAME;
    }

    public String[] getPropertyNames() {
        return PROPERTY_NAMES;
    }
}

File: src/main/kotlin/no/nav/pensjon/innsyn/domain/desc/OpptjeningDescriptor.kt
package no.nav.pensjon.innsyn.domain.desc

import no.nav.pensjon.innsyn.domain.Opptjening
import no.nav.pensjon.innsyn.entity.EntityDescriptor

object OpptjeningDescriptor : EntityDescriptor<Opptjening>("Opptjening",
        arrayOf(
                "Type",
                "Status",
                "Opptjeningsår",
                "Pensjonsgivende inntekt",
                "Poeng",
                "Uføregrad"
        )
)